stargz/stargz.go

```diff
@@ -762,6 +774,9 @@ func (w *Writer) AppendTar(r io.Reader) error {
 		} else {
 			w.toc.Entries = append(w.toc.Entries, ent)
 		}
+		if regFileEntry != nil {
+			regFileEntry.Checksum = fmt.Sprintf("sha256:%x", payloadChecksum.Sum(nil))
```

Comment:
Why does it store checksum per file? `nodeHandle.Read` deal with file payload per chunk rather than per file at `crfs.go`.

Reply:
An issue with using chunks for the checksum is that the chunk size can be tweaked by the writer and it is not constant, so I don't think we cannot depend on that.

The use case I've in mind is that the cache is handled separately from crfs itself.  When the file is in the cache, I'd skip completely "OpenFile".

Reply:
Yes, ChunkSize isn't consistent, but each ChunkSize is stored in each TOCEntry so the filesystems can deal with the correspondences of digests and ChunkSizes correctly?(sorry if you considering other concern, if so please  tell me ;)

Your assumed cache implementation is nice, but currently CRFS doesn't have it. And isn't it better to keep wide range of possibility for the cache implementations by simply adding per-chunk digest *not only* per-file digest?(I personally think READ-driven caching/fetching is also common cache implementation not only your OPEN-driven one)

Reply:
I think we could have a per-chunk checksum as well, that would still be different than the file checksum I am adding.

I've not considered the per-chunk checksum as I currently don't need/use it.  Should we add once there is at least one user that is going to need it?  @bradfitz what do you think?

Reply:
@giuseppe 
> Should we add once there is at least one user that is going to need it?
Yes, I agree with it!